clientcontroller/evm_consumer.go

```diff
@@ -81,27 +86,167 @@ func (ec *EVMConsumerController) QueryFinalityProviderVotingPower(fpPk *btcec.Pu
 }
 
 func (ec *EVMConsumerController) QueryLatestFinalizedBlocks(count uint64) ([]*types.BlockInfo, error) {
-	return ec.queryLatestBlocks(nil, count, finalitytypes.QueriedBlockStatus_FINALIZED, true)
+
+	lastnumber, err := ec.GetLatestFinalizedNumber()
+	if err != nil {
+		return nil, fmt.Errorf("can't get latest finalized block:%s", err)
+
+	}
+	type Block struct {
+		Number string
+		Hash   string
+	}
+
+	var B []rpc.BatchElem
+
+	for i := 0; i < int(count); i++ {
+
+		hexStr := Transform(lastnumber)
+		ib := rpc.BatchElem{
+			Method: "eth_getBlockByNumber",
+			Args:   []interface{}{hexStr, true},
+			Result: new(Block),
+		}
+		B = append(B, ib)
+		lastnumber.Sub(lastnumber, big.NewInt(1))
+	}
```

Comment:
`QueryLatestFinalizedBlocks` has same code. for readability and maintanence, we should try to minimize code duplicate as much as we can.

if you are using a code snippet in two places, extract it to helper function.

also avoid writing large functions. keep it modular by splitting code to smaller functions.

Reply:
Add a new function InitBatch(Batch []rpc.BatchElem, number *big.Int, count uint64, order string) to avoid duplicates.